### PR TITLE
DOCS-4937 fix

### PR DESCRIPTION
**Pull request by george-hoffer**
https://www.mulesoft.org/jira/browse/DOCS-4937 -- "Object store documentation is misleading"

### CODE REVIEW

File: modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -48,11 +48,10 @@ image::os-use-object-store-v2.png[Click Use Object Store V2 in the Runtime Manag
 * The Object Store Connector works for both Object Store version 1 and 2.
 * Object Store v2 enforces a maximum data persistence of 30 days from time of creation.
 * Accounts that have not purchased the Object Store v2 premium SKU are rate limited to 10 transactions per second per application. To go beyond the base 10 TPS rate limit, contact your MuleSoft account team.
-* In Mule 4, Object Store v2 overrides the `_defaultPersistentObjectStore` object store. No further configuration is required.
+* Object Store v2 overrides the `_defaultPersistentObjectStore` object store when creating Mule 4 applications. No further configuration is required.
```
Comment (fededonna) [inline modules/ROOT/pages/index.adoc]: We should make clear that this only happens in CH

File: modules/ROOT/pages/index.adoc [hunk 2]
```diff
@@ -48,11 +48,10 @@ image::os-use-object-store-v2.png[Click Use Object Store V2 in the Runtime Manag
...
 If the connector or component that you are using uses that object store, then you have access to Object Store v2.
 +
-In Mule 3, Object Store v2 overrides the `_defaultUserObjectStore` object store. In the Mule 3 Object Store connector,
-you must set the Object Store Reference field to `_defaultUserObjectStore` for data to be stored in Object Store v2.
+When creating Mule 3 applications, Object Store v2 overrides the `_defaultUserObjectStore` object store. In the Mule 3 Object Store connector, you must set the Object Store Reference field to `_defaultUserObjectStore` for data to be stored in Object Store v2.
```
Comment (fededonna) [inline modules/ROOT/pages/index.adoc]: Same here

File: modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -10,7 +10,7 @@ to write directly to object store, use its REST API, and write and read using
 the Object Store connector. See the xref:osv2-faq.adoc[FAQ] for
 Object Store information.
 
-NOTE: Mule 4 only supports Object Store v2.
+NOTE: CloudHub only supports Object Store v2 for creating Mule 4 applications. This does not affect on-premises use of Mule 4. Object Store v2 only stores data in the cloud. On-premises use of Mule stores data locally to a persistent data store.
```
Comment (fededonna) [inline modules/ROOT/pages/index.adoc]: CH supports osv2 and in memory os.

**Review by fededonna — CHANGES_REQUESTED** [review]

**Comment by steveCluster** [comment]
I agree with Fede's comments.

File: modules/ROOT/pages/index.adoc (hunk 1 above)
Comment (george-hoffer) [reply modules/ROOT/pages/index.adoc]: Changed wording to:
`* For cloud applications using Mule 4, Object Store v2 overrides the _defaultPersistentObjectStore object store. No further configuration is required. If a cloud connector or component uses that object store, then you have access to Object Store v2.`

`For cloud applications using Mule 3, Object Store v2 overrides the _defaultUserObjectStore object store. To use Object Store v2 in cloud applications, set the Object Store Reference field to the _defaultUserObjectStore value.`

File: modules/ROOT/pages/index.adoc [hunk 4]
```diff
@@ -48,11 +49,9 @@ image::os-use-object-store-v2.png[Click Use Object Store V2 in the Runtime Manag
 * The Object Store Connector works for both Object Store version 1 and 2.
 * Object Store v2 enforces a maximum data persistence of 30 days from time of creation.
 * Accounts that have not purchased the Object Store v2 premium SKU are rate limited to 10 transactions per second per application. To go beyond the base 10 TPS rate limit, contact your MuleSoft account team.
-* In Mule 4, Object Store v2 overrides the `_defaultPersistentObjectStore` object store. No further configuration is required.
-If the connector or component that you are using uses that object store, then you have access to Object Store v2.
+* For cloud applications using Mule 4, Object Store v2 overrides the `_defaultPersistentObjectStore` object store. No further configuration is required. If a cloud connector or component uses that object store, then you have access to Object Store v2.
```
Comment (steveCluster) [inline modules/ROOT/pages/index.adoc]: `If a cloud connector or component uses that object store, then you have access to Object Store v2.` should be `If a cloud connector or component uses that object store, then the data will be persisted in Object Store v2.`

File: modules/ROOT/pages/index.adoc (hunk 3 above)
Comment (steveCluster) [reply modules/ROOT/pages/index.adoc]: Yes options are OSv2 and in-memory for Mule 4.

**Review by steveCluster — APPROVED** [review]

**Review by fededonna — APPROVED** [review]

File: modules/ROOT/pages/index.adoc [hunk 5]
```diff
@@ -2,15 +2,16 @@
...
+Object Store v2 lets CloudHub applications store data and states across batch processes
 and Mule components, from within a single application. Object Store v2 acts as the
-backend for Mule 3 and Mule 4 object store access, which enables applications
+backend for Mule 3 and Mule 4 CloudHub object store access, which enables applications
 to write directly to object store, use its REST API, and write and read using
 the Object Store connector. See the xref:osv2-faq.adoc[FAQ] for
 Object Store information.
 
-NOTE: Mule 4 only supports Object Store v2.
+NOTE: CloudHub applications that use Mule 4 can enable use of Object Store v2 or in-memory Object Store. On-premises Mule 4 applications do not use Object Store v2, but instead store data locally to a persistent data store. Object Store v2 can only be used in CloudHub.
```
Comment (valkyrie69) [inline modules/ROOT/pages/index.adoc]: CloudHub applications that use Mule 4 can enable the use of Object Store v2? Or you can use Object Store v2 with CloudHub apps running on Mule 4? 
I would put the last sentence first.

File: modules/ROOT/pages/index.adoc [hunk 6]
```diff
@@ -48,11 +49,9 @@ image::os-use-object-store-v2.png[Click Use Object Store V2 in the Runtime Manag
...
 * Object Store v2 enforces a maximum data persistence of 30 days from time of creation.
 * Accounts that have not purchased the Object Store v2 premium SKU are rate limited to 10 transactions per second per application. To go beyond the base 10 TPS rate limit, contact your MuleSoft account team.
-* In Mule 4, Object Store v2 overrides the `_defaultPersistentObjectStore` object store. No further configuration is required.
-If the connector or component that you are using uses that object store, then you have access to Object Store v2.
+* For CloudHub applications using Mule 4, if you choose Object Store v2, Object Store v2 overrides the `_defaultPersistentObjectStore` object store.
 +
-In Mule 3, Object Store v2 overrides the `_defaultUserObjectStore` object store. In the Mule 3 Object Store connector,
-you must set the Object Store Reference field to `_defaultUserObjectStore` for data to be stored in Object Store v2.
+For CloudHub applications using Mule 3, Object Store v2 overrides the `_defaultUserObjectStore` object store. To use Object Store v2 in CloudHub applications, set the Object Store Reference field to the `_defaultUserObjectStore` value.
```
Comment (valkyrie69) [inline modules/ROOT/pages/index.adoc]: I don't understand this. 
Object Store v2 overrides the `_defaultUserObjectStore` object store reference on CloudHub apps running on Mule 3? 
Second sentence doesn't specify CloudHub apps running on Mule 3, but Im assuming you need to set that object store reference only for Mule 3 apps? Or do you need to do that in all cases if you are using Object Store v2?

**Review by valkyrie69 — APPROVED** [review]

**Pull request merged**